crates/tabby/src/main.rs

```diff
@@ -54,19 +56,35 @@ pub enum Commands {
     /// Execute the repository sync job.
     #[cfg(feature = "ee")]
     #[clap(name = "job::sync")]
-    JobSync,
+    JobSync(JobArgs),
 
     /// Execute the index job.
     #[cfg(feature = "ee")]
     #[clap(name = "job::index")]
-    JobIndex,
+    JobIndex(JobArgs),
+}
+
+#[derive(clap::Args)]
+pub struct JobArgs {
+    #[clap(long, requires = "address")]
+    token: Option<String>,
+    #[clap(long, requires = "port")]
+    addr: Option<String>,
+    #[clap(long, requires = "token")]
+    port: Option<u16>,
```

Comment:
there's no need to have addr / port. Please just have single argument of `url`.

it's similar to https://github.com/TabbyML/tabby/blob/5dc3303f73a60beb5867f1e242f88d509fc44606/crates/tabby/src/worker.rs#L24